syntax = "proto3";

package Message;

message Message {
  enum Type {
    REQUEST = 0;
    PROPOSAL = 1;
    DECISION = 2;
  }
  Type messageType = 1;
  oneof payload {
    RequestMessage req = 2;
    ProposalMessage pro = 3;
    DecisionMessage dec = 4;
  }
}


message RequestMessage {
  optional int32 userId = 1;
  int32 requestId = 2;
  string query = 3;
}

message ProposalMessage {
  int32 slotId = 1;
  RequestMessage command = 2;
}

message DecisionMessage {
  int32 slotId = 1;
  RequestMessage command = 2;
}
